### ISSUE TITLE
Strict concurrency checking warns when using withTaskCancellationHandler in a global actor context

### ISSUE THREAD

**Issue by Jnosh** [apple/swift#61119]
**Describe the bug**
When using `withTaskCancellationHandler` in a global actor context (e.g. `@MainActor`), strict concurrency checking issues a warning that the operation closure is non-sendable and cannot cross actor boundaries.

**Steps To Reproduce**

Minimal example that reproduces the warning:

```swift
@MainActor
func play() async {
    await withTaskCancellationHandler { // Non-sendable type '() async throws -> ()' exiting main actor-isolated context in call to non-isolated global function 'withTaskCancellationHandler(operation:onCancel:)' cannot cross actor boundary
    
    } onCancel: {

    }
}
```

**Expected behavior**
It should be possible to use `withTaskCancellationHandler` in a global actor context.

Perhaps `withTaskCancellationHandler` should allow running the operation in the same global actor context it is located in or maybe there should be a version accepting a  `@Sendable` operation closure if it is necessary to transfer the operation across actors.

**Environment**
 - OS: macOS 13.0 Beta (22A5342f)
 - Xcode Version/Tag/Branch: Version 14.1 beta (14B5024h)

**Comment by ktoso** [comment]
This is likely related to rdar://99845352 where the same is the case in normal actors

**Comment by Jnosh** [comment]
This seems to be fixed as of Swift 5.9, thanks!
